js/main.js

```diff
@@ -52,7 +65,65 @@ function getResults(searchRequest) {
       buyLink.textContent = 'Buy Now'
       buyLink.className = 'buy-column'
       result.appendChild(buyLink);
+      var buyButton = document.querySelectorAll('.buy-column');
+      var gameId = result.setAttribute('gameid', this.response[i].gameID);
+      var gameTitle = result.setAttribute('game-title', this.response[i].external);
+      for (var j = 0; j < buyButton.length; j++) {
+        buyButton[j].addEventListener('click', buyNow)
+      }
     }
   });
   search.send()
 }
+
+function buyNow(event) {
+  function removeAllChildNodes(parent) {
+    while (parent.firstChild) {
+      parent.removeChild(parent.firstChild);
+    }
+  }
+  removeAllChildNodes(storeListings);
+  searchResults.className = "hidden";
+  var gameIdResult = this.parentNode.getAttribute("gameid");
+  var prices = new XMLHttpRequest();
+  prices.open("GET", "https://www.cheapshark.com/api/1.0/games?id=" + gameIdResult)
+  prices.responseType = 'json';
+  prices.addEventListener('load', function () {
+    for (var i = 0; i < this.response.deals.length; i++) {
+      storeListings.className = "search-results";
+      var resultHeader = gameTitle;
+      console.log(resultHeader)
+      var priceResult = document.createElement('li');
+      priceResult.className = "result-row"
+      storeListings.appendChild(priceResult);
+      var storeId = this.response.deals[i].storeID;
+      priceResult.setAttribute("storeid", storeId);
+      var storeIcon = document.createElement('img');
+      var iconImage;
+      for(var k = 0; k < storesList.length; k++) {
+        if (storeId === storesList[k].storeID) {
+          iconImage = "https://www.cheapshark.com" + storesList[k].images['banner'];
+        }
+      }
+      storeIcon.setAttribute('src', iconImage);
+      storeIcon.className = 'list-image picture-column';
+      priceResult.appendChild(storeIcon);
+      var storeName = document.createElement('h2');
+      var storeActualName;
+      for (var j = 0; j < storesList.length; j++) {
+        if (storeId === storesList[j].storeID) {
+          storeActualName = storesList[j].storeName;
+        }
+      }
```

Comment:
Rather than looping over the store list twice, you should be able to do this in the above loop too 👍